backend/hdfs_backend.go

```diff
@@ -42,20 +45,34 @@ func (h *HdfsBackend) LatestVersion(checkForSuccess bool) (string, error) {
 	return "", fmt.Errorf("No valid versions at %s", h.displayURL(h.path))
 }
 
-func (h *HdfsBackend) Download(version string, destPath string) error {
+func (h *HdfsBackend) Download(version string, destPath string) (rterr error) {
 	versionPath := path.Join(h.path, version)
 	files, err := h.client.ReadDir(versionPath)
 	if err != nil {
 		return err
 	}
 
+	// To avoid loading an incomplete download (#12), download into a temp dir
+	// then rename the temp dir to destPath only if all downloads succeed.
+	baseDir := path.Dir(destPath)
+	workDir, err := ioutil.TempDir(baseDir, fmt.Sprintf("version-%v", version))
```

Comment:
Good idea! I'll give it a try.

On Thu, May 7, 2015 at 5:52 PM, Colin Marc <notifications@github.com> wrote:

> In backend/hdfs_backend.go
> <https://github.com/stripe/sequins/pull/13#discussion_r29899714>:
>
> >  	versionPath := path.Join(h.path, version)
> >  	files, err := h.client.ReadDir(versionPath)
> >  	if err != nil {
> >  		return err
> >  	}
> >
> > +	// To avoid loading an incomplete download (#12), download into a temp dir
> > +	// then rename the temp dir to destPath only if all downloads succeed.
> > +	baseDir := path.Dir(destPath)
> > +	workDir, err := ioutil.TempDir(baseDir, fmt.Sprintf("version-%v", version))
>
> I think we can push this up into the toplevel type, and pass the directory
> as destPath here. Does that work?
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/stripe/sequins/pull/13/files#r29899714>.
>